app/public/utils/url/buildUrl.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+/**
+ * @typedef {string|number|null|boolean} QueryParameterValue
+ */
+
+/**
+ * Build a URL from a base URL (that may already have query parameters) and a list of query parameters
+ *
+ * @param {string} baseURL the base URL to which parameters should be added
+ * @param {object} parameters the query parameters
+ * @return {string} URL the built URL
+ */
+exports.buildUrl = (baseURL, parameters) => {
+    if (!parameters) {
+        parameters = {};
+    }
+    const [url, existingParameters] = baseURL.split('?');
```

Comment:
I think it is better to use a `class URL` in general, instead of operating on raw strings.

Reply:
that's how it's done in the bookkeeping